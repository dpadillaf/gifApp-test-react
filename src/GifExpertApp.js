import { useState } from "react/cjs/react.development";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";
import React from 'react';


const GifExpertApp = () => {

    const [categories, setCategories] = useState([ 'One Punch' ]);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr />

            <ol>
                {
                    categories.map( category => (
                       <GifGrid 
                            key={ category }
                            category={ category } 
                        />
                    ) )
                }
            </ol>
        </>
    );
}

export default GifExpertApp;