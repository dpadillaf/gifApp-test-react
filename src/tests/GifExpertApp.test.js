import { shallow } from 'enzyme'
import GifExpertApp from '../GifExpertApp'

describe('Test in GifExpertApp', () => {
    
    test('muestra GifExpertApp correctamente', () => {
        
        const wrapper = shallow( <GifExpertApp /> )
        expect( wrapper ).toMatchSnapshot();
        
    })

    test('debe mostrar una lista de categorias', () => {
        
        const categories = [ 'snk', 'one piece' ];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'GifGrid' ).length ).toBe( categories.length );

    })
    

})
