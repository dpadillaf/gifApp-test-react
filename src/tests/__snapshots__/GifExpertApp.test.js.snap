// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifExpertApp debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="snk"
      key="snk"
    />
    <GifGrid
      category="one piece"
      key="one piece"
    />
  </ol>
</Fragment>
`;

exports[`Test in GifExpertApp muestra GifExpertApp correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
